bridge/engine/EngineMixer.cpp

```diff
@@ -4014,143 +4061,148 @@ void EngineMixer::onBarbellUserMediaMap(size_t barbellIdHash, const char* messag
     auto videoEndpointsArray = mediaMapJson["video-endpoints"].getArray();
     auto audioEndpointsArray = mediaMapJson["audio-endpoints"].getArray();
 
-    memory::Array<BarbellMapItem, 12> videoSsrcs;
-    memory::Array<BarbellMapItem, 8> audioSsrcs;
-
+    memory::Map<size_t, BarbellMapItem, 16> videoSsrcs;
     copyToBarbellMapItemArray(videoEndpointsArray, videoSsrcs);
-    copyToBarbellMapItemArray(audioEndpointsArray, audioSsrcs);
-
-    memory::Map<size_t, size_t, 32> fwdVideoEndpoints;
-    for (auto& item : videoSsrcs)
+    for (auto& stream : barbell->videoStreams)
     {
-        fwdVideoEndpoints.add(item.endpointIdHash, item.endpointIdHash);
+        if (stream.endpointIdHash.isSet())
+        {
+            addToMap(stream, videoSsrcs);
+        }
     }
-
-    memory::Map<size_t, size_t, 32> fwdAudioEndpoints;
-    for (auto& item : audioSsrcs)
+    if (barbell->slideStream.endpointIdHash.isSet())
```

Comment:
I think store `slideStream` separated of `videoStreams` gave us nothing than confusion. We are used to handle slideStreams as part of videoStreams for the real users and this separation already cause some problems by forgetting to process this stream.

Wouldn't it be better add it to videoStreams instead?

Reply:
The reasons was to be able to identify which stream is slides without searching as it is special and may not be used by video. But it seems we do not use that yet.